pkg/pool-weighted/test/foundry/CircuitBreakerLib.t.sol

```diff
@@ -0,0 +1,144 @@
+    uint256 private constant _MAXIMUM_TOKEN_WEIGHT = 99e16; // 0.99 (99%)
+    uint256 private constant _MAX_BOUND_PERCENTAGE = 2e18; // 2.0
+    uint256 private constant _MIN_BPT_PRICE = 1e6;
+
+    uint256 private constant _MAX_RELATIVE_ERROR = 1e16;
+    uint256 private constant _MAX_BPT_PRICE = type(uint112).max;
+
+    function testReferenceParamsAndBoundRatios(
+        uint256 bptPrice,
+        uint256 weightComplement,
+        uint256 lowerBound,
+        uint256 upperBound
+    ) public {
+        bptPrice = bound(bptPrice, _MIN_BPT_PRICE, _MAX_BPT_PRICE);
+        weightComplement = bound(weightComplement, _MINIMUM_TOKEN_WEIGHT, _MAXIMUM_TOKEN_WEIGHT);
+        lowerBound = bound(lowerBound, _MINIMUM_BOUND_PERCENTAGE, FixedPoint.ONE);
+        upperBound = bound(upperBound, FixedPoint.ONE, _MAX_BOUND_PERCENTAGE);
+
+        CircuitBreakerLib.CircuitBreakerParams memory params = CircuitBreakerLib.CircuitBreakerParams({
+            bptPrice: bptPrice,
+            weightComplement: weightComplement,
+            lowerBound: lowerBound,
+            upperBound: upperBound
+        });
+
+        // The setter overwrites all state, so the previous state doesn't matter
+        // If we find we need to set fields individually (e.g., only the bounds),
+        // we could add tests that the previous state was not altered.
+        bytes32 poolState = CircuitBreakerLib.setCircuitBreakerFields(params);
+        CircuitBreakerLib.CircuitBreakerParams memory result = CircuitBreakerLib.getCircuitBreakerFields(poolState);
+
+        assertEq(result.bptPrice, bptPrice);
+        assertEq(result.weightComplement, weightComplement);
+        assertApproxEqRel(result.lowerBound, lowerBound, _MAX_RELATIVE_ERROR);
+        assertApproxEqRel(result.upperBound, upperBound, _MAX_RELATIVE_ERROR);
+
+        bytes32 initialPoolState = CircuitBreakerLib.setCircuitBreakerFields(params);
+        (uint256 initialLowerBptPriceBoundary, uint256 initialUpperBptPriceBoundary) =
+            CircuitBreakerLib.getCurrentCircuitBreakerBounds(initialPoolState, weightComplement);
+
+        uint256 expectedLowerBoundBptPrice = uint256(bptPrice).mulDown(lowerBound.powDown(weightComplement));
+        uint256 expectedUpperBoundBptPrice = uint256(bptPrice).mulDown(upperBound.powUp(weightComplement));
+
+        assertApproxEqRel(initialLowerBptPriceBoundary, expectedLowerBoundBptPrice, _MAX_RELATIVE_ERROR);
+        assertApproxEqRel(initialUpperBptPriceBoundary, expectedUpperBoundBptPrice, _MAX_RELATIVE_ERROR);
+
+        // Test that calling it with the original weightComplement retrieves exact values from the ratio cache
+        (uint256 cachedLowerBptPriceBoundary, uint256 cachedUpperBptPriceBoundary) =
+            CircuitBreakerLib.getCurrentCircuitBreakerBounds(initialPoolState, weightComplement);
+
+        assertEq(cachedLowerBptPriceBoundary, initialLowerBptPriceBoundary);
+        assertEq(cachedUpperBptPriceBoundary, initialUpperBptPriceBoundary);
```

Comment:
This should be a standalone test, not included in the other one. The first part tests the encoding, while this tests the computation of the bounds when the weight complement has not changed.